### PR TITLE
Move flaky tests to the flakytests folder/group

### PR DESCRIPTION
**Pull request by kinow**
This is a small change with no associated Issue.

See https://github.com/cylc/cylc-flow/issues/2894#issuecomment-524670221 for context. These tests failed in a build in Travis CI for a PR, but passed on Hilary's fork Travis CI. They also passed for me locally.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).

- [x] No change log entry required (why? e.g. invisible to users).

- [x] No documentation update required.

### CODE REVIEW

**Comment by kinow** [comment]
Hmm, Travis failed on my fork already, in the unit-test stage due to shellcheck?? Strange as I just moved the files.

```
$ .travis/shellchecker

In flakytests/registration/02-on-the-fly.t line 39:

cd "${TESTD}"

^-----------^ SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

For more information:

  https://www.shellcheck.net/wiki/SC2164 -- Use 'cd ... || exit' or 'cd ... |...

The command ".travis/shellchecker" exited with 1.
```

**Comment by kinow** [comment]
Alright, so there was an exclusion in shellchecker for `tests`, but no exclusion for `flakytests`.

I've included an extra commit in this PR to address that.

**Comment by kinow** [comment]
Last failure was in the sensitive tests stage in Travis. Whew. Will kick it now until Travis is green again.

**Comment by kinow** [comment]
Done, and I considering the total time since the kick off of the build, until the sensitive tests passed:  Total time 2 hrs 35 min 10 sec 

image

**Comment by hjoliver** [comment]
How many kicks needed?

**Comment by kinow** [comment]
Just 1

**Comment by kinow** [comment]
Oh, sorry. Fixed (edited commit). Going to take a few more minutes now for Travis to process this last change 👍

File: flakytests/registration/test_header [hunk 1]
```diff
@@ -0,0 +1 @@
+../lib/bash/test_header
```
Comment (kinow) [inline flakytests/registration/test_header]: Oh, interesting. Never noticed how symlinks are displayed in GitHub diff UI.

Here's my local copy:

image

So hopefully created the symlink correctly.

**Comment by matthewrmshin** [comment]
Note: In moving tests to live in flakytests/, I've modified some source files to ensure that they pass in shellcheck normal mode, hence there is no exclusion for that directory. I'm preparing a branch to do the same for the rest of the source files living under the tests/ directory. I've just got it working at the end of Friday.

Happy to merge this first. I'll take a good look tomorrow. It is public holiday in the UK today.

**Comment by kinow** [comment]
@matthewrmshin I would feel bad for adding a problem in flakytests, now that I know you had the trouble to fix it :slightly_smiling_face: 

Updated the commit to know keep shellchecker running on `flakytests`, but tried to fix the issue it reported earlier today... it was about `cd` into a directory. Here's the output from running it locally:

```bash
In flakytests/registration/02-on-the-fly.t line 39:
cd "${TESTD}"
^-- SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
```

Not really fluent in Shell... but the commit now uses `cd "${..}" || exit 1`... which I hope it a valid fix? Otherwise just let me know if there's a better way and I'll update the PR :+1: 

Thanks!

**Comment by matthewrmshin** [comment]
Travis CI failure will be fixed by #3319.

**Comment by hjoliver** [comment]
Did you want to merge this before yours @matthewrmshin?

**Pull request merged**